### PR TITLE
django 1.6 compatability

### PR DESCRIPTION
**Pull request by snopoke**
@dimagi/team-commcare-hq 

Django 1.5 is no longer getting bugfixes or security updates. Time to upgrade.

The following changes are required to upgrade to django 1.6. They are all backwards compatible with django 1.5.

### CODE REVIEW

File: corehq/apps/domain/urls.py [hunk 1]
```diff
@@ -69,16 +69,20 @@ def extend(d1, d2):
     ) +\
     patterns('django.contrib.auth.views',
         url(r'^accounts/password_change/$', 'password_change', auth_pages_path('password_change_form.html'), name='password_change'),
-        url(r'^accounts/password_change_done/$', 'password_change_done', auth_pages_path('password_change_done.html') ),
+        url(r'^accounts/password_change_done/$', 'password_change_done', auth_pages_path('password_change_done.html'),
```
Comment (dimagimon) [inline corehq/apps/domain/urls.py]: E501 line too long (118 > 115 characters)

File: corehq/apps/domain/urls.py [hunk 2]
```diff
@@ -69,16 +69,20 @@ def extend(d1, d2):
...
+            name='password_change_done'),
 
         url(r'^accounts/password_reset_email/$', exception_safe_password_reset, extend(auth_pages_path('password_reset_form.html'),
                                                                                        { 'password_reset_form': ConfidentialPasswordResetForm,
                                                                                          'from_email':settings.DEFAULT_FROM_EMAIL}),
-                                                                                name='password_reset_email'),
-        url(r'^accounts/password_reset_email/done/$', 'password_reset_done', auth_pages_path('password_reset_done.html') ),
+            name='password_reset_email'),
+        url(r'^accounts/password_reset_email/done/$', 'password_reset_done', auth_pages_path('password_reset_done.html'),
```
Comment (dimagimon) [inline corehq/apps/domain/urls.py]: E501 line too long (121 > 115 characters)

File: corehq/apps/domain/urls.py [hunk 3]
```diff
@@ -69,16 +69,20 @@ def extend(d1, d2):
...
+            name='password_reset_done'),
 
-        url(r'^accounts/password_reset_confirm/(?P<uidb36>[0-9A-Za-z]+)-(?P<token>.+)/$', 'password_reset_confirm', auth_pages_path('password_reset_confirm.html'), name="confirm_password_reset" ),
-        url(r'^accounts/password_reset_confirm/done/$', 'password_reset_complete', auth_pages_path('password_reset_complete.html') ) 
+        url(r'^accounts/password_reset_confirm/(?P<uidb36>[0-9A-Za-z]+)-(?P<token>.+)/$', 'password_reset_confirm',
+            auth_pages_path('password_reset_confirm.html'), name="confirm_password_reset" ),
```
Comment (dimagimon) [inline corehq/apps/domain/urls.py]: E202 whitespace before ')'

File: corehq/apps/domain/urls.py [hunk 4]
```diff
@@ -69,16 +69,20 @@ def extend(d1, d2):
...
+        url(r'^accounts/password_reset_confirm/done/$', 'password_reset_complete', auth_pages_path('password_reset_complete.html'),
```
Comment (dimagimon) [inline corehq/apps/domain/urls.py]: E501 line too long (131 > 115 characters)

**Comment by twymer** [comment]
Maybe it's on the ambitious side, but I assume we should target getting on 1.7 if we're going to put time into making an upgrade (which we should certainly do). Last time we pushed for 1.5 and then within a year we were once again on an unsupported version.

Build fail looks real but unrelated, going to rerun anyway.

**Comment by snopoke** [comment]
Yea, 1.7 has quite a few more changes. But feel free to take a look and see
if you think its doable.

**Comment by snopoke** [comment]
build failure fixed by https://github.com/dimagi/commcare-hq/pull/5272

**Comment by NoahCarnahan** [comment]
@snopoke 
Looks like the `django-markup-deprecated` item in `requirements.txt` needs a version spec. I get an error when I run pip install.

**Comment by snopoke** [comment]
@NoahCarnahan odd, I don't get any error

**Comment by NoahCarnahan** [comment]
Yeah... maybe something else is going on. I just tried adding a version
number but I still get the same error. Looking into it some more.

**Comment by NoahCarnahan** [comment]
doh. Looks like I just forgot the `-r` flag.

**Comment by esoergel** [comment]
@twymer I'm also very much in favor of staying up-to-date with the latest releases, especially since Django [doesn't seem keen](https://docs.djangoproject.com/en/1.7/internals/release-process/#long-term-support-lts-releases) to do more long-term support releases:

**Pull request merged**